syntax = "proto3";

package tkrzw;

message StatusProto {
  uint32 code = 1;
  string message = 2;
}

message StringPair {
  string first = 1;
  string second = 2;
}

message GetVersionRequest {
}

message GetVersionResponse {
  string version = 1;
}

message ReportRequest {
}

message ReportResponse {
  repeated StringPair records = 1;
}

message GetRequest {
  uint32 db_index = 1;
  bytes key = 2;
}

message GetResponse {
  StatusProto status = 1;
  bytes value = 2;
}

message SetRequest {
  uint32 db_index = 1;
  bytes key = 2;
  bytes value = 3;
  bool overwrite = 4;
}

message SetResponse {
  StatusProto status = 1;
}

message RemoveRequest {
  uint32 db_index = 1;
  bytes key = 2;
}

message RemoveResponse {
  StatusProto status = 1;
}

service DBMService {
  rpc GetVersion(GetVersionRequest) returns (GetVersionResponse);
  //rpc Report(ReportRequest) returns (ReportResponse);
  //rpc Get(GetRequest) returns (GetResponse);
  //rpc Set(SetRequest) returns (SetResponse);
  //rpc Remove(GetRequest) returns (RemoveResponse);
}
